src/components/admin/form-components/withSelector.jsx

```diff
@@ -0,0 +1,177 @@
+        ) !== undefined;
+
+      if (alreadyAdded) {
+        await this.props.displayAlert(
+          `The ${this.props.mode} ${title} has already been added.`,
+        );
+        return;
+      }
+      const newObjects = update(this.props.elements, {
+        $push: [{ id, name: title, slug }],
+      });
+      this.props.onUpdate(newObjects);
+    };
+
+    handleDelete(id, title, slug) {
+      // disabled this if saving
+      if (this.props.disabled) return;
+
+      const objectSlug = !slug ? this.props.slugify(title) : slug;
+
+      const index = this.props.elements.findIndex(
+        baseObject =>
+          !('slug' in baseObject)
+            ? this.props.slugify(baseObject.name) === objectSlug
+            : baseObject.slug === objectSlug,
+      );
+      if (index === -1) {
+        throw new Error(
+          'You tried to delete an object not currently selected. ' +
+            "This shouldn't happen, please let the developers know that it did.",
+        );
+      }
+      const newObjects = update(this.props.elements, { $splice: [[index, 1]] });
+      this.props.onUpdate(newObjects);
+    }
+
+    render() {
+      const styles = {
+        wrapper: {
+          display: 'flex',
+          flexWrap: 'wrap',
+        },
+      };
+
+      const objectChips =
+        this.props.elements.length > 0
+          ? _.map(this.props.elements, object => (
+              <ObjectChip
+                key={object.id !== null ? object.id : object.name}
+                id={object.id}
+                title={object.name}
+                onDelete={this.handleDelete}
+                style={{ margin: 4 }}
+              >
+                {object.name}
+              </ObjectChip>
+            ))
+          : null;
+
+      const noObjectsMessage = (
+        <span style={{ color: 'rgba(0, 0, 0, 0.3)' }}>
+          No {this.props.mode} are currently assigned to this article
+        </span>
+      );
+
+      return (
+        <div>
+          <br />
+          <p style={{ marginTop: 0, marginBottom: 10 }}>
+            {capFirstLetter(this.props.mode)}
+          </p>
+          <div style={styles.wrapper}>{objectChips || noObjectsMessage}</div>
+          <WrappedField
+            length={3}
+            handleClick={this.handleClickAdd}
+            {...this.props}
+          />
+        </div>
+      );
+    }
+  }
+
+  Selector.propTypes = {
+    elements: PropTypes.arrayOf(
+      PropTypes.shape({
+        id: PropTypes.number,
+        name: PropTypes.string.isRequired,
+      }),
+    ).isRequired,
+    onChange: PropTypes.func.isRequired,
+    onUpdate: PropTypes.func.isRequired,
+    displayAlert: PropTypes.func.isRequired,
+    slugify: PropTypes.func.isRequired,
+    // It isn't a typo and they are required in SearchBar so no need for defaults
+    /* eslint-disable react/no-typos, react/require-default-props */
+    mode: React.PropTypes.oneOf(['staff', 'articles']).isRequired,
```

Comment:
We don't want to use the deprecated `React.PropTypes` anymore (especially when the rest of them don't use it), just use the library, I think that may also be the reason for the typo linting error?

Reply:
Also, you can just do `// eslint-disable-next-line react/require-default-props` instead of enabling and disabling